src/renderers/WebGLRenderer.js

```diff
@@ -2462,6 +2462,12 @@ function WebGLRenderer( parameters ) {
 
 	}() );
 
+	this.setTexture2DArray = function ( texture, slot ) {
```

Comment:
Does this need to be added to `WebGLRenderer`?

Reply:
It's called from `WebGLUniforms`.

Reply:
Ah, I see. We should deprecate these functions and move all that code to `WebGLTextures` at some point...

Reply:
And pass `textures` to `WebGLUniforms` instead.

Reply:
Would you like to do that clean up?
Otherwise I'll put it on my list (unless @Mugen87 wants to take care of it).

Reply:
According to the document, `WebGLRenderer.setTexture*()` seem to be expected being not only called from `WebGLUniforms` but also directly called from users. 

https://threejs.org/docs/#api/en/renderers/WebGLRenderer.setTexture2D

Reply:
Do you think it's ok to remove these methods from renderer? It may break user code. (I'm not sure how user use these methods.)

Reply:
Should we make a new issue for this?

Reply:
> Do you think it's ok to remove these methods from renderer? 

We probably should not remove `setTexture`, `setTexture2D` nor `setTextureCube` yet, but I think we should remove the ones we have added recently `setTexture3D` and `setTexture2DArray`. Then change `WebGLUniforms` so it uses `WebGLTextures` directly.

> Should we make a new issue for this? I'm not sure how these methods are used in the code base right now.

Sounds good to me.

Reply:
> Sounds good to me.

#15924 🎉